### PR TITLE
Mapper proposal

### PR DESCRIPTION
**Pull request by sailorvii**
**What type of PR is this?**
 /kind design

**What this PR does / why we need it**:
Simplify the mapper structure

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
All mappers will be refined.

### CODE REVIEW

**Comment by luogangyi** [comment]
@kevin-wangzefeng @fisherxu let's discuss it in next community meeting

File: docs/proposals/mapper-redesign.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+last-updated: 
+---
+
+# Mapper Re-design
+## Introduction
+
+The original description could refer to the doc [mapper-design.md](https://github.com/kubeedge/kubeedge/blob/master/docs/proposals/mapper-design.md "mapper-design.md").
+
+Mapper is an interface between kubeedge and devices. It could set/get device data, get and report the device status.
+Kubeedge uses device controller, device twin and mapper to control the devices.
```
Comment (kevin-wangzefeng) [inline docs/proposals/mapper-redesign.md:20]:
```suggestion
KubeEdge uses device controller, device twin and mapper to control the devices. 
```

File: docs/proposals/mapper-redesign.md [hunk 2]
```diff
@@ -0,0 +1,139 @@
+---
+title: Mapper re-design
+status: implementable
+authors:
+    - "@sailorvii"
+    - "@luogangyi"
+approvers:
+  - "@"
+  - "@"
+creation-date: 2020-09-
```
Comment (kevin-wangzefeng) [inline docs/proposals/mapper-redesign.md:10]:
```suggestion
approvers:
  - "@kevin-wangzefeng"
  - "@"
creation-date: 2020-09-09
```

File: docs/proposals/mapper-redesign.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+### Miscellaneous
+#### Device status definition
+```go
+const DeviceStatus {
+    DEVSTOK
+    DEVSTERR        /*Expected value is not equal as setting*/
+    DEVSTDISCONN    
+    DEVSTUNHEALTHY  /*Unhealthy status from device*/
+}
```
Comment (kevin-wangzefeng) [inline docs/proposals/mapper-redesign.md:138]: Not quite understand the code here, what's the value type of the consts?

**Comment by kevin-wangzefeng** [comment]
Minor comments, thanks

File: docs/proposals/mapper-redesign.md (hunk 3 above)
Comment (sailorvii) [reply docs/proposals/mapper-redesign.md:138]: Just want to present all status that reported to the device controller. In the code, the type is uint.

File: docs/proposals/mapper-redesign.md (hunk 1 above)
Comment (sailorvii) [reply docs/proposals/mapper-redesign.md]: Thanks for your review and time.

Done.

**Comment by kevin-wangzefeng** [comment]
nit: maybe "mapper refactoring proposal" or "mapper design v2" is better for the proposal name

**Comment by fisherxu** [comment]
@sailorvii Let's rename the proposal to "mapper design v2"？

File: docs/proposals/mapper-design-v2.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    - "@sailorvii"
+    - "@luogangyi"
+approvers:
+  - "@kevin-wangzefeng"
+  - "@"
+creation-date: 2020-09-09
+last-updated: 
+---
+
+# Mapper Re-design
```
Comment (fisherxu) [inline docs/proposals/mapper-design-v2.md:14]:
```suggestion
# Mapper Design V2
 ```

**Review by fisherxu — COMMENTED** [review]
Also please squash the commits :)

**Pull request merged**